src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureKeyVaultReference/AzureKeyVaultKeyValueAdapter.cs

```diff
@@ -102,5 +92,34 @@ public bool NeedsRefresh()
         {
             return _secretProvider.ShouldRefreshKeyVaultSecrets();
         }
+
+        private string ParseSecretReferenceUri(ConfigurationSetting setting)
+        {
+            try
+            {
+                var reader = new Utf8JsonReader(System.Text.Encoding.UTF8.GetBytes(setting.Value));
+
+                if (reader.Read() && reader.TokenType == JsonTokenType.StartObject)
+                {
+                    if (reader.Read() && reader.TokenType == JsonTokenType.PropertyName && reader.GetString() == KeyVaultConstants.SecretReferenceUriJsonPropertyName)
```

Comment:
Extra properties shouldn't break deserialization.

Reply:
Got it, so read all property names until we see the uri if it exists

Reply:
yes